### PR TITLE
Correcting instantiation of dispatches vector

### PR DESCRIPTION
**Pull request by jj16791**

### CODE REVIEW

**Review by FinnWilkinson — APPROVED** [review]
Looks fine to me

**Comment by dANW34V3R** [comment]
A short PR description would be good explaining the problem and fix for future reference

**Comment by rahahahat** [comment]
Looks good to me

**Review by rahahahat — APPROVED** [review]

File: src/lib/pipeline/DispatchIssueUnit.cc [hunk 1]
```diff
@@ -56,8 +56,8 @@ void DispatchIssueUnit::tick() {
 
   /** Stores the number of instructions dispatched for each
    * reservation station. */
-  std::vector<uint16_t> dispatches = {
-      0, static_cast<unsigned short>(reservationStations_.size())};
+  std::vector<uint16_t> dispatches(
+      static_cast<unsigned short>(reservationStations_.size()), 0);
```
Comment (dANW34V3R) [inline src/lib/pipeline/DispatchIssueUnit.cc:60]: No cast needed

**Review by dANW34V3R — CHANGES_REQUESTED** [review]

File: src/lib/pipeline/DispatchIssueUnit.cc (hunk 1 above)
Comment (jj16791) [reply src/lib/pipeline/DispatchIssueUnit.cc:60]: Had warnings on the MacStudio for it not being cast so will be keeping casting.

Comment (dANW34V3R) [reply src/lib/pipeline/DispatchIssueUnit.cc:60]: What was the warning?

Comment (jj16791) [reply src/lib/pipeline/DispatchIssueUnit.cc:60]: Wasn't recorded.

**Comment by jj16791** [comment]
> https://github.com/UoB-HPC/SimEng/pull/257#issuecomment-1291976452

Not needed as it's self-explanatory.

**Review by dANW34V3R — APPROVED** [review]

**Pull request merged**